src/idl_gen_cpp.cpp

```diff
@@ -878,6 +870,18 @@ class CppGenerator : public BaseGenerator {
         code += name + "Identifier()); }\n\n";
       }
 
+      // The root verifier:
+      code += "inline bool Verify";
+      code += name;
+      code +=
+          "Buffer(flatbuffers::Verifier &verifier) { "
+          "return ";
+      if (parser_.file_identifier_.length())
+          code += "verifier.GetBufferSize() > (2 * sizeof(flatbuffers::uoffset_t)) && " + name + "BufferHasIdentifier(verifier.GetBuffer()) && ";
```

Comment:
Maybe wrap this logic into `Verifier::VerifyIdentifier` or something, so this doesn't need to go into all generated code. Also if you modify generated code, make sure to run `tests\generate_code.sh` to make the generated code changes part of your commit